### ISSUE TITLE
Modal Overlay scroll blocking is not compatible with the overflow-y scroll style of html elements 

### ISSUE THREAD

**Issue by hungtcs** [ant-design/ant-design#51503]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-jyh2k9)

### Steps to reproduce

1. Set html style `overflow-y: scroll`
2. Open Modal with scroll bars 

### What is expected?

1. Scrollbar is hidden
2. Keep current view position unchanged

### What is actually happening?

1. First the scroll bar flashes
2. Then the view position is reset to the top

| Environment | Info |
| --- | --- |
| antd | 5.21.6 |
| React | 18.3.1 |
| System | MacOS 13.3 (22E252) |
| Browser | Google Chrome 130.0.6723.92（正式版本） (arm64) |

**Comment by hungtcs** [comment]
In Angular CDK their solution looks like this

<img width="2237" alt="image" src="https://github.com/user-attachments/assets/28a5d22d-a674-4435-8513-8b07b3570dd9">

<https://stackblitz.com/edit/q47nwv?file=.gitignore,src%2Findex.html,src%2Fstyles.scss>

If refer to the angular cdk solution for optimization, and combine it with the current body width minus the scroll bar width, it should be the most complete solution at present.

**Comment by MadCcc** [comment]
Remove `overflow-y` on html and seems everything is ok.

**Comment by hungtcs** [comment]
> Remove `overflow-y` on html and seems everything is ok.

Yes, it may be because the browser has some special behavior after setting `overflow` on the html element. 
But my purpose is to statically display a scroll bar on html, so I can't just remove `overflow-y` to solve this problem.
